scripts/util.js

```diff
@@ -0,0 +1,40 @@
+const path = require('path');
+const shelljs = require('shelljs');
+
+/** Map of common typos in target names. The key is the typo, the value is the correct form. */
+const commonTypos = new Map([
+  ['snackbar', 'snack-bar'],
+]);
+
+// List of packages where the specified component could be defined in. The script uses the
+// first package that contains the component (if no package is specified explicitly).
+// e.g. "button" will become "material/button", and "overlay" becomes "cdk/overlay".
+const orderedGuessPackages = ['material', 'cdk', 'material-experimental', 'cdk-experimental'];
+
+/**
+ * Tries to guess the full name of a package, based on a shorthand name.
+ * Returns an object with the result of the guess and the names that were attempted.
+ */
+function guessPackageName(name, packagesDir) {
+  // Try to correct the name if it's in the list of common typos.
+  name = commonTypos.get(name) || name;
```

Comment:
Before the typo's were corrected anywhere in the passed name. e.g. I could pass `yarn test material/snackbar`. Can we preserve this?

Reply:
Maybe it was possible, but I don't think it was used since the only mapping we have right not is `snackbar` to `snack-bar` which is still covered by these changes.

Reply:
I think the typo's will be only corrected if just the typo is the argument, but I'm fine leaving as is because we don't have a lot of typo's captured here. I'm not sure how people pass in `snackbar` anyway, but it seems like it could have been `yarn test material/snackbar`: #18770

Reply:
I see what you mean now: since the replacement was going through `String.prototype.replace`, it would correct things like `materila/snackbar` too. I've put the old logic back in.